pir/cpp/server.h

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2020 the authors listed in CONTRIBUTORS.md
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef PIR_SERVER_H_
+#define PIR_SERVER_H_
+
+#include <string>
+#include <vector>
+
+#include "context.h"
+#include "seal/seal.h"
+#include "util/statusor.h"
+
+namespace pir {
+
+using ::private_join_and_compute::StatusOr;
+
+class PIRServer {
+ public:
+  /**
+   * Creates and returns a new server instance.
+   * @returns TODO
+   **/
+  static StatusOr<std::unique_ptr<PIRServer>> Create();
```

Comment:
Does it make sense to populate a server with no params and no database? Perhaps we should restrict this to actually passing in valid values?